unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/PrimitiveMesh/AssetPromise_PrimitiveMesh.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections;
+using DCL.Helpers;
+using UnityEngine;
+
+namespace DCL
+{
+    public class AssetPromise_PrimitiveMesh: AssetPromise<Asset_PrimitiveMesh>
+    {
+        public PrimitiveMeshModel model;
+
+        private IPrimitiveMeshFactory primitiveMeshFactory;
+
+        public AssetPromise_PrimitiveMesh(PrimitiveMeshModel model)
+        {
+            this.model = model;
+            primitiveMeshFactory = new PrimitiveMeshFactory();
```

Comment:
why are we creating a instance of this on every asset promise?

Reply:
This is due to encapsulation but lets wait for @AjimenezDCL  to see what he think

Reply:
why would we need encapsulation here if it's a factory and not part of the object state?